File: advent-of-code/2023/one/one.py
import os


def read_file(file_path: str) -> list[str]:
    with open(file_path, "r") as file:
        f = file.readlines()
    return f


def numbers(line: str) -> int: 
    first_number, second_number= '0','0'
    for c in line: 
        if c.isnumeric():
            first_number = c
            break 
    for c in line[::-1]: 
        if c.isnumeric():
            second_number = c
            break 
    return int(first_number + second_number)


if __name__ == "__main__": 

    dir = os.path.dirname(os.path.abspath(__file__))
    lines = read_file(os.path.join(dir, 'lines.txt'))
    print(sum(map(numbers, lines)))

File: advent-of-code/2023/one/one.js
const fs = require("fs");
const path = require("path");

const readlines = (filename) => {
  let filepath = path.join(__dirname, filename);
  return fs
    .readFileSync(filepath, "utf8", (err, data) => {
      if (err) {
        console.error(err);
        return;
      }
      return data;
    })
    .trim();
};

let filedata = readlines("lines.txt");
let lines = filedata.split("\n");
let sum = 0;

for (let j = 0; j < lines.length; j++) {
  let first_char;
  let last_char;
  let line = lines[j];

  for (let i = 0; i < line.length; i++) {
    if (!isNaN(parseInt(line[i]))) {
      first_char = line[i];
      break;
    }
  }

  for (let k = line.length - 1; k >= 0; k--) {
    if (!isNaN(parseInt(line[k]))) {
      last_char = line[k];
      break;
    }
  }

  let number = parseInt(first_char + last_char);
  sum += number;
}

console.log(sum);
